### PR TITLE
doc: Explain rebase policy in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by maflcko**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -149,6 +149,20 @@ the respective change set.
...
+
+This project aims to have a clean git history, where code changes are only made in non-merge commits. This simplifies
+auditability because merge commits can be assumed to not contain arbitrary code changes. Merge commits should be signed,
+and the resulting git tree hash must be deterministic and reproducible. The script in
+[/contrib/verify-commits](/contrib/verify-commits) checks that.
+
+After a rebase, reviewers are encouraged to sign off on the force push. This should be relatively straightforward with
+the `git range-diff` tool explained in the [productivity
+notes](/doc/productivity.md#diff-the-diffs-with-git-range-diff). To avoid needless review churn, maintainers will
+generally merge pull request that received the most review attention first.
```
Comment (jonatack) [inline CONTRIBUTING.md]: slrequest/requests/

**Review by jonatack — COMMENTED** [review]
ACK modulo typo

File: CONTRIBUTING.md (hunk 1 above)
Comment (MarcoFalke) [reply CONTRIBUTING.md]: Thanks, done!

**Comment by sanjaykdragon** [comment]
Unrelated, but what is the point of squashing commits?

**Comment by MarcoFalke** [comment]
@sanjaykdragon Explained in the latest force push

**Comment by fanquake** [comment]
Is this an alternative to #17940?

**Comment by MarcoFalke** [comment]
@fanquake Wasn't planned, but I see that they are mostly overlapping.

**Comment by emilengler** [comment]
ACK faecaf82233e37be876f76356e9dc69a03c2c340

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -119,10 +119,10 @@ to your fork until you have satisfied all feedback.
 
 Note: Code review is a burdensome but important part of the development process, and as such, certain types of pull requests are rejected. In general, if the **improvements** do not warrant the **review effort** required, the PR has a high chance of being rejected. It is up to the PR author to convince the reviewers that the changes warrant the review effort, and if reviewers are "Concept NAK'ing" the PR, the author may need to present arguments and/or do research backing their suggested changes.
 
-Squashing Commits
----------------------------
-If your pull request is accepted for merging, you may be asked by a maintainer
-to squash and or [rebase](https://git-scm.com/docs/git-rebase) your commits
+### Squashing Commits
+
+If your pull request contains fixup commits (commits that change the same line of code repeatetly) or too fine-grained
```
Comment (hebasto) [inline CONTRIBUTING.md]: typo?
s/repeatetly/repeatedly/

**Review by hebasto — COMMENTED** [review]
ACK faecaf82233e37be876f76356e9dc69a03c2c340 modulo typo.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -119,10 +119,10 @@ to your fork until you have satisfied all feedback.
 
 Note: Code review is a burdensome but important part of the development process, and as such, certain types of pull requests are rejected. In general, if the **improvements** do not warrant the **review effort** required, the PR has a high chance of being rejected. It is up to the PR author to convince the reviewers that the changes warrant the review effort, and if reviewers are "Concept NAK'ing" the PR, the author may need to present arguments and/or do research backing their suggested changes.
```
Comment (jonatack) [inline CONTRIBUTING.md]: while fixing up the typo (repeatetly), perhaps also:

s/"Concept NAK'ing"/"Concept NACK" on/

**Comment by MarcoFalke** [comment]
Fixed typos as requested by @jonatack and @hebasto

**Comment by hebasto** [comment]
re-ACK fa1244783ccb4798af4906ac6be5a38df241fa38, typos fixed.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -149,6 +149,20 @@ the respective change set.
 The length of time required for peer review is unpredictable and will vary from
 pull request to pull request.
 
+### Rebasing Changes
+
+When a pull request conflicts with the target branch, you may be asked to rebase it on top of the current target branch.
+The `git rebase` command will take care of rebuilding your commits on top of the new base.
+
+This project aims to have a clean git history, where code changes are only made in non-merge commits. This simplifies
+auditability because merge commits can be assumed to not contain arbitrary code changes. Merge commits should be signed,
```
Comment (jonatack) [inline CONTRIBUTING.md]: maybe s/auditability/auditing/ but doesn't really matter

**Review by jonatack — COMMENTED** [review]
ACK

**Comment by laanwj** [comment]
ACK fa1244783ccb4798af4906ac6be5a38df241fa38

**Pull request merged**